### PR TITLE
Adding Multi-Region Network Deployment

### PR DESCRIPTION
**Pull request by brsteph**
## Overview/Summary

This PR adds the ability to a networking stack into two regions as part of the ALZ portal deployment.

## This PR fixes/adds/changes/removes

1. Deploy multiple Hub virtual networks, and have them peered together
2. Associate single DDoS Network Protection instance to all hub networks deployed
3. Deploy Private DNS Zones for Azure PaaS services to multiple regions
4. Associate Private DNS Zones to all virtual networks
5. Deploy VPN Gateway to multiple regions
6. Deploy ExpressRoute Gateway to multiple regions
7. Deploy Azure Firewall to multiple regions
8. Deploy Azure Firewall policies to support multi-region **REVISITING**
9. Deploy vWAN Virtual Hub to multiple regions, including embedded services
10. Deploy Identity virtual network in multiple regions

### Breaking Changes

1. Unsure at this time - may need feedback on this.

## Testing Evidence

Please provide any testing evidence to show that your Pull Request works/fixes as described and planned (include screenshots, if appropriate).

image
image
image

(More can be provided as needed)

### Testing URLs

The below URLs can be updated where the placeholders are, look for `{YOUR GITHUB BRANCH NAME HERE - Remove Curly Brackets Also}` & `{YOUR GITHUB BRANCH NAME HERE - Remove Curly Brackets Also}`, to allow you to test your portal deployment experience.

> Please also replace the curly brackets on the placeholders `{}`

#### Azure Public

[Deploy To Azure](https://portal.azure.com/#view/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fbrsteph%2FEnterprise-Scale%2Fmain%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2Fbrsteph%2FEnterprise-Scale%2Fmain%2FeslzArm%2Feslz-portal.json)

#### Azure US Gov (Fairfax)
[Deploy To Azure](https://portal.azure.us/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fbrsteph%2FEnterprise-Scale%2Fmain%22FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2Fbrsteph%2FEnterprise-Scale%2Fmain%2FeslzArm%2Ffairfaxeslz-portal.json)

## As part of this Pull Request I have

- [X] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [X] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [X] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [X] Performed testing and provided evidence.
- [X] Ensured [contribution guidance](https://github.com/Azure/Enterprise-Scale/wiki/ALZ-Contribution-Guide) is followed.
- [X] Updated relevant and associated documentation.
- [X] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

### CODE REVIEW

File: docs/wiki/Deploying-ALZ-HubAndSpoke.md [hunk 1]
```diff
@@ -74,12 +74,39 @@ Depending on your requirements, you may choose to deploy additional network infr
 
  ![img](./media/clip_image036b.png)
 
+### Deploying networking resources in an additional region
```
Comment (paulgrimley) [inline docs/wiki/Deploying-ALZ-HubAndSpoke.md:77]: should we use the language deploying to multiple regions? I appreciate for portal this is restricted to 2 regions so perhaps for this the language works well, @jtracey93 thoughts?

File: docs/wiki/Deploying-ALZ-HubAndSpoke.md [hunk 2]
```diff
@@ -74,12 +74,39 @@ Depending on your requirements, you may choose to deploy additional network infr
...
-On the *Identity* blade you can specify if you want to assign recommended policies to govern identity and domain controllers. If you decide to enable this feature, you do need to provide an empty subscription for this. You can then select which policies you want to get assigned, and you will need to provide the address space for the virtual network that will be deployed on this subscription. Please note that this virtual network will be connected to the hub virtual network via VNet peering. 
+On the *Identity* blade you can specify if you want to assign recommended policies to govern identity and domain controllers. If you decide to enable this feature, you do need to provide an empty subscription for this. You can then select which policies you want to get assigned, and you will need to provide the address space for the virtual network that will be deployed on this subscription. Please note that this virtual network will be connected to the hub virtual network via VNet peering.
 
  ![img](./media/clip_image036c.png)
 
+In addition, if you wish to deploy an additional virtual network in a secondary region, if you are deploying network resources in the secondary region.
```
Comment (paulgrimley) [inline docs/wiki/Deploying-ALZ-HubAndSpoke.md:106]: this doesnt read well and not clear on what is trying to be said

File: docs/wiki/Deploying-ALZ-VWAN.md [hunk 3]
```diff
@@ -69,12 +69,38 @@ Depending on your requirements, you may choose to deploy additional network infr
 
 ![vwan](./media/clip_image078.jpg)
 
+### Deploying networking resources in an additional region
+
+You may also deploy an additional virtual hub in an additional region, to prepare for multi-region operations.
```
Comment (paulgrimley) [inline docs/wiki/Deploying-ALZ-VWAN.md:74]: should we be stronger in our wording of this being our recommended approach - i personally think so

**Review by paulgrimley — CHANGES_REQUESTED** [review]
Reviewed the documentation part and need some updates

File: docs/wiki/Deploying-ALZ-VWAN.md (hunk 3 above)
Comment (brsteph) [reply docs/wiki/Deploying-ALZ-VWAN.md]: Updated in the commit, let me know your thoughts!

File: docs/wiki/Deploying-ALZ-HubAndSpoke.md (hunk 1 above)
Comment (jtracey93) [reply docs/wiki/Deploying-ALZ-HubAndSpoke.md]: i say leave as is, as it matches the RI option

**Review by jtracey93 — APPROVED** [review]
LGTM 🚀

**Pull request merged**